packages/hoppscotch-common/src/helpers/oauth.ts

```diff
@@ -309,4 +296,32 @@ const clearPKCEState = () => {
   persistenceService.removeLocalConfig("client_secret")
 }
 
+async function runRequestThroughInterceptor(config: AxiosRequestConfig) {
+  if (interceptorService.currentInterceptorID.value === "extension") {
+    // wait till the extension resolves
+    await until(extensionService.extensionStatus).toMatch(
+      (status) => status !== "waiting",
+      {
+        timeout: 1000,
+      }
+    )
+  }
```

Comment:
I would recommend to keep specific Interceptor logic out of the general app code.

What if we do this wait within `ExtensionInterceptorService`'s `runRequest` call ?

Reply:
yeah. that would work.